### PR TITLE
Getting scores out of beam_ner

### PR DESCRIPTION
**Pull request by svlandeg**
Adding back the `get_beam_parses` method for the NER & added a `scored_ents` function. Not entirely sure this is how we want to do this?

TODO: parser (I'll do this in a separate PR)

## Description
Enabling the test for Issue https://github.com/explosion/spaCy/issues/4313 again - unfortunately currently broken again.

### Types of change
bug fix

## Checklist
- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/tests/regression/test_issue4001-4500.py [hunk 1]
```diff
@@ -288,35 +288,33 @@ def set_annotations(self, docs, scores):
...
+    }
+    ner = nlp.add_pipe("beam_ner", config=config)
     ner.add_label("SOME_LABEL")
-    ner.initialize(lambda: [])
+    nlp.initialize()
     # add a new label to the doc
     doc = nlp("What do you think about Apple ?")
     assert len(ner.labels) == 1
     assert "SOME_LABEL" in ner.labels
+    ner.add_label("MY_ORG")   # TODO: not sure if we want this to be necessary...
```
Comment (svlandeg) [inline spacy/tests/regression/test_issue4001-4500.py:307]: I think this shouldn't be necessary: when we add a previously trained NER to a pipeline with a matcher, you'd need to add the matcher's labels to the NER. Ideally, this test should work without this line.

Comment (honnibal) [reply spacy/tests/regression/test_issue4001-4500.py:307]: On the other hand the resize-while-running thing is pretty quirky. We have a separate discussion about that, so I'll leave this alone for here.

**Comment by honnibal** [comment]
This looks right to me, go ahead and merge after that conflict is resolved 👍

File: spacy/tests/regression/test_issue4001-4500.py (hunk 1 above)
Comment (svlandeg) [reply spacy/tests/regression/test_issue4001-4500.py:307]: Ok! I've created a quick ticket as a reminder: https://github.com/explosion/spaCy/issues/6681

**Pull request merged**